karpenter/datadog_checks/karpenter/config_models/validators.py

```diff
@@ -0,0 +1,40 @@
+# (C) Datadog, Inc. 2023-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+from urllib.parse import urlparse
+
+
+def initialize_instance(values, **kwargs):
+    if 'openmetrics_endpoint' in values:
+        validate_url(
+            values['openmetrics_endpoint'],
+            required_path='/metrics',
+            example='http://localhost:8000/metrics',
+            config='openmetrics_endpoint',
+        )
+
+    return values
+
+
+def validate_url(url, required_path=None, example=None, config=None):
```

Comment:
note for later, not relevant to this particular PR: I think it would be worth it to have this in the base check

Reply:
Yeah I think I can move this over to the base check since I ended up using it more than once.